pkg/envoy/sort.go

```diff
@@ -83,51 +83,52 @@ type PortNetworkPolicyRuleSlice []*cilium.PortNetworkPolicyRule
 
 // PortNetworkPolicyRuleLess reports whether the r1 rule should sort before
 // the r2 rule.
+// L3-L4-only rules are less than L7 rules.
 func PortNetworkPolicyRuleLess(r1, r2 *cilium.PortNetworkPolicyRule) bool {
-	remotePolicies1, remotePolicies2 := r1.RemotePolicies, r2.RemotePolicies
+	// TODO: Support Kafka.
+
+	http1, http2 := r1.GetHttpRules(), r2.GetHttpRules()
 	switch {
-	case len(remotePolicies1) < len(remotePolicies2):
+	case http1 == nil && http2 != nil:
 		return true
-	case len(remotePolicies1) > len(remotePolicies2):
+	case http1 != nil && http2 == nil:
```

Comment:
Should we be handling
```
 case http1 == nil && http2 == nil:
 return false
```
I guess it is a nop in that case, so probably doesn't matter?

Reply:
Or we need to make it fall through to check remote policies in this case?

Reply:
This is already handled here: https://github.com/cilium/cilium/pull/3510/files#diff-f96d05f217d54d321e99d0ff16cbc0a1R98